internal/expr/expr_test.go

```diff
@@ -510,6 +529,26 @@ func (s *ExprSuite) TestPrepareErrors(c *C) {
 		query:       "SELECT &NoTags.* FROM t",
 		prepareArgs: []any{NoTags{}},
 		err:         `cannot prepare statement: output expression: no "db" tags found in struct "NoTags": &NoTags.*`,
+	}, {
+		query:       "SELECT street FROM t WHERE x = $Address.*",
+		prepareArgs: []any{Person{}, Manager{}, Address{}},
+		err:         `cannot prepare statement: input expression: star type not allowed in standalone input: $Address.*`,
```

Comment:
"star type",  "standalone input", ... it feels like some of these concepts are not really part of the set of ideas we need to convey to get the system to work properly, and they are naturally somewhat alien to developers that are used to either Go or SQL, so the less we need to explain, the better.

How about replacing most (all?) of these with the more boring:

```
cannot prepare statement: "$Address.*" used in invalid context
```
?

Reply:
I agree that "standalone input" is not needed here. I would like to keep the "star" or "asterisk" though because here it is the offending part of the statement and I think its worth bringing the users attention to that. I've changed it to:
```
cannot prepare statement: input expression: asterisk used in invalid context: $Address.*
```